packages/dev/core/src/Materials/Node/Blocks/Fragment/imageProcessingBlock.ts

```diff
@@ -156,16 +156,16 @@ export class ImageProcessingBlock extends NodeMaterialBlock {
             if (color.connectedPoint!.type === NodeMaterialBlockConnectionPointTypes.Color4 || color.connectedPoint!.type === NodeMaterialBlockConnectionPointTypes.Vector4) {
                 state.compilationString += `${state._declareOutput(output)} = ${color.associatedVariableName};\n`;
             } else {
-                state.compilationString += `${state._declareOutput(output)} = vec4(${color.associatedVariableName}, 1.0);\n`;
+                state.compilationString += `${state._declareOutput(output)} = vec4${state.fSuffix}(${color.associatedVariableName}, 1.0);\n`;
             }
             state.compilationString += `#ifdef IMAGEPROCESSINGPOSTPROCESS\n`;
             if (this.convertInputToLinearSpace) {
-                state.compilationString += `${output.associatedVariableName}.rgb = toLinearSpace(${color.associatedVariableName}.rgb);\n`;
+                state.compilationString += `${output.associatedVariableName} = vec4${state.fSuffix}(toLinearSpaceVec3(${color.associatedVariableName}.rgb), ${color.associatedVariableName}.a);\n`;

```

Comment:
Don't we also need a `toLinearSpaceVec3` in WebGL? It seems this function only exists in `helperFunctions.fx` for WGSL, not for WebGL.

Reply:
nice catch!!